### PR TITLE
chore: move to named exports

### PR DESCRIPTION
**Pull request by montezume**
#### Summary

Instead of importing from `@commercetools-frontend/x/something` we only import from `@commercetools-frontend/x` and use named exports.

Also, removed the mix of default and named exports in `application-shell`. [ref](https://github.com/rollup/rollup/issues/1961).

This will help us on our transition towards shipping bundled code.

(This will be a breaking change due to the removing default ApplicationShell export)

### CODE REVIEW

File: packages/application-shell/src/index.js [hunk 1]
```diff
@@ -3,7 +3,8 @@
  */
 import './public-path';
 
-export { default } from './components/application-shell';
```
Comment (emmenko) [inline packages/application-shell/src/index.js]: Can we keep it the `default`? I consider this to be the **main** component of the package and the rest are additional/complementary stuff that gets exported, so there is like a distinction of which component is more important to use.

But it's not super important if there are good reasons to refactor it...

File: packages/application-shell/src/index.js [hunk 2]
```diff
@@ -3,7 +3,8 @@
...
+export { default as ApplicationShell } from './components/application-shell';
+export { default as apolloClient } from './configure-apollo';
```
Comment (emmenko) [inline packages/application-shell/src/index.js]: Was this exposed on purpose?

File: packages/i18n/async-locale-data/async-locale-data.spec.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import { shallow } from 'enzyme';
 import { reportErrorToSentry } from '@commercetools-frontend/sentry';
-import loadI18n from '@commercetools-frontend/i18n/load-i18n';
```
Comment (emmenko) [inline packages/i18n/async-locale-data/async-locale-data.spec.js]: Thanks, not sure what happened here 🤦‍♂️

File: packages/application-shell/src/index.js (hunk 1 above)
Comment (montezume) [reply packages/application-shell/src/index.js]: I guess we could keep both for now. But in general, it's not a great idea to expose both a default and named exports from a single module

https://github.com/rollup/rollup/issues/1961

File: packages/application-shell/src/index.js (hunk 2 above)
Comment (montezume) [reply packages/application-shell/src/index.js]: yes, it's used in the discounts app.

File: packages/application-shell/src/index.js (hunk 1 above)
Comment (emmenko) [reply packages/application-shell/src/index.js]: Ah I see, it's mostly for CJS support. Alright, it's fine by me then 💯

Comment (tdeekens) [reply packages/application-shell/src/index.js]: I would not reccomend that either. I also do not mind a full and painful refactor across all apps with this maybe even not being backwards compat. It should be a do-able effort and a long standing refactor passed onto teams might be in sum a bigger effort than a at once version.

**Comment by emmenko** [comment]
Maybe we should have waited before merging this. It's a breaking change and I was thinking of releasing this together with the rollup bundle to only have one breaking release.

I guess we won't do any more releases in between...

**Pull request merged**